spec/models/dialog_spec.rb

```diff
@@ -82,18 +82,27 @@
   end
 
   context "#destroy" do
+    let(:dialog) { FactoryBot.create(:dialog) }
+
     it "destroy without resource_action association" do
-      expect(FactoryBot.create(:dialog).destroy).to be_truthy
+      expect(dialog.destroy).to be_truthy
       expect(Dialog.count).to eq(0)
     end
 
-    it "destroy with resource_action association" do
-      dialog = FactoryBot.create(:dialog)
+    it "does not destroy dialog with resource_action association" do
       FactoryBot.create(:resource_action, :action => "Provision", :dialog => dialog, :resource_type => Dialog, :resource_id => dialog.id)
-      expect { dialog.destroy }
-      .to raise_error(RuntimeError, "Dialog cannot be deleted because it is connected to other components: [\"Dialog:#{dialog.id} - #{dialog.name}\"]")
+      dialog.destroy
+      err_msg = dialog.errors.full_messages[0]
+      expect(err_msg).to eq("Dialog cannot be deleted because it is connected to other components: [\"Dialog:#{dialog.id} - #{dialog.name}\"]")
       expect(Dialog.count).to eq(1)
     end
+
+    describe "#reject_if_has_resource_actions" do
+      it "before_destroy callback throws :abort if dialog has associatex resource" do
+        FactoryBot.create(:resource_action, :action => "Provision", :dialog => dialog, :resource_type => Dialog, :resource_id => dialog.id)
+        expect { dialog.send(:reject_if_has_resource_actions) }.to throw_symbol(:abort)
```

Comment:
I don't think we need to test the private method itself, the test above covers the use case, right?